infrastructure/testdriver/bidi/bluetooth/resources/bidi-bluetooth-helper.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+/**
+ * Waits until the document has finished loading.
+ * @returns {Promise<void>} Resolves if the document is already completely
+ *     loaded or when the 'onload' event is fired.
+ */
+function waitForDocumentReady() {
+  return new Promise(resolve => {
+    if (document.readyState === 'complete') {
+      resolve();
+    }
+
+    window.addEventListener('load', () => {
+      resolve();
+    }, {once: true});
+  });
+}
+
+/**
+ * Simulates a user activation prior to running |callback|.
+ * @param {Function} callback The function to run after the user activation.
+ * @returns {Promise<*>} Resolves when the user activation has been simulated
+ *     with the result of |callback|.
+ */
+async function callWithTrustedClick(callback) {
+  await waitForDocumentReady();
+  return new Promise(resolve => {
+    let button = document.createElement('button');
+    button.textContent = 'click to continue test';
+    button.style.display = 'block';
+    button.style.fontSize = '20px';
+    button.style.padding = '10px';
+    button.onclick = () => {
+      document.body.removeChild(button);
+      resolve(callback());
+    };
+    document.body.appendChild(button);
+    test_driver.click(button);
+  });
+}
+
+/**
+ * Register a one-time handler that selects the first device in the device
+ * prompt upon a device prompt updated event.
+ */
+function selectFirstDeviceOnDevicePromptUpdated() {
+  test_driver.bidi.bluetooth.request_device_prompt_updated.once().then(
```

Comment:
The same concern with potential unhandled promise and test timing out.

Reply:
Done.